datafusion/common/src/dfschema.rs

```diff
@@ -605,25 +582,49 @@ impl DFSchema {
     }
 
     /// Returns true if the two schemas have the same qualified named
-    /// fields with the same data types. Returns false otherwise.
+    /// fields with the compatible data types. Returns false otherwise.
     ///
     /// This is a specialized version of Eq that ignores differences
     /// in nullability and metadata.
     ///
     /// Use [DFSchema]::logically_equivalent_names_and_types for a weaker
     /// logical type checking, which for example would consider a dictionary
     /// encoded UTF8 array to be equivalent to a plain UTF8 array.
-    pub fn equivalent_names_and_types(&self, other: &Self) -> bool {
+    pub fn equivalent_names_and_types(&self, other: &Self) -> Result<()> {
```

Comment:
likewise, to ease upgrade, can you please put the old function signature back and mark it deprecated and put the new signature as a new function? Perhaps something like

```rust
   #[deprecated(since = "47.0.0", note = "Use has_equivalent_names_and_types`  instead")]
    pub fn equivalent_names_and_types(&self, other: &Self) -> bool {
      self.has_equivalent_names_and_types.is_err()
    }

    pub fn has_equivalent_names_and_types(&self, other: &Self) -> Result<()> { 
...
    }
```

Reply:
Thank you @alamb for view, got it, good guide for me to upgrade. Addressed in latest PR.